torchtitan/components/quantization/float8.py

```diff
@@ -26,9 +26,15 @@ def __init__(self, job_config: JobConfig, parallel_dims: ParallelDims):
         self.enabled = False
 
         float8_config: Float8 = job_config.float8
-        if not has_cuda_capability(8, 9):
+        if not has_cuda_capability(8, 9) and not float8_config.emulate:
             logger.warning(
-                "Failed to swap to Float8Linear because float8 is only supported on SM89 or later",
+                "Failed to swap to Float8Linear because float8 is only supported on SM89 or later."
+                "To enable support on older hardware, set `float8.emulate` to True.",
+            )
+            return
+        elif float8_config.emulate and job_config.training.compile:
+            logger.warning(
+                "Failed to run on emulate with compile on, please disable compile to allow on emulate.",
```

Comment:
We should just raise an exception if the configurations combination is not runnable.